spec/src/lib.rs

```diff
@@ -387,13 +393,44 @@ impl SystemCells {
         self.files.len()
     }
 
-    fn build_outputs(&self) -> Result<(Vec<packed::CellOutput>, Vec<Bytes>), Box<dyn Error>> {
+    fn build_outputs(
+        &self,
+        input: &packed::CellInput,
+        output_index_start: u64,
+    ) -> Result<(Vec<packed::CellOutput>, Vec<Bytes>), Box<dyn Error>> {
         let mut outputs = Vec::with_capacity(self.files.len());
         let mut outputs_data = Vec::with_capacity(self.files.len());
-        for res in &self.files {
+        for (offset, res) in self.files.iter().enumerate() {
             let data: Bytes = res.get()?.into_owned().into();
+            let data_hash = packed::CellOutput::calc_data_hash(&data).pack();
+
+            let type_script = if [
```

Comment:
It's better to configure whether the cell should have the type script, instead of hard coding here.

Suggested config

```
[[genesis.system_cells]]
file =  { bundled = "specs/cells/secp256k1_blake160_sighash_all" }
create_type_id = true

...

[genesis.system_cells_lock]
code_hash = "0xb35557e7e9854206f7bc13e3c3a7fa4cf8892c84a09237fb0aab40aab3771eee"
args = []
hash_type = "Data"
```

Reply:
ok

Reply:
@doitian 
Integration test report `'chain spec serialize: ValueAfterTable'`.
It's a issue by serialize complex data structure to toml.